src/ImportDefinitionsBundle/Interpreter/IteratorInterpreter.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Import Definitions.
+ *
+ * LICENSE
+ *
+ * This source file is subject to the GNU General Public License version 3 (GPLv3)
+ * For the full copyright and license information, please view the LICENSE.md and gpl-3.0.txt
+ * files that are distributed with this source code.
+ *
+ * @copyright  Copyright (c) 2016-2018 w-vision AG (https://www.w-vision.ch)
+ * @license    https://github.com/w-vision/ImportDefinitions/blob/master/gpl-3.0.txt GNU General Public License version 3 (GPLv3)
+ */
+
+namespace ImportDefinitionsBundle\Interpreter;
+
+use CoreShop\Component\Registry\ServiceRegistryInterface;
+use ImportDefinitionsBundle\Model\DataSetAwareInterface;
+use ImportDefinitionsBundle\Model\DataSetAwareTrait;
+use ImportDefinitionsBundle\Model\DefinitionInterface;
+use ImportDefinitionsBundle\Model\Mapping;
+use Pimcore\Model\DataObject\Concrete;
+use Webmozart\Assert\Assert;
+
+final class IteratorInterpreter implements InterpreterInterface, DataSetAwareInterface
+{
+    use DataSetAwareTrait;
+
+    /**
+     * @var ServiceRegistryInterface
+     */
+    private $interpreterRegistry;
+
+    /**
+     * @param ServiceRegistryInterface $interpreterRegistry
+     */
+    public function __construct(ServiceRegistryInterface $interpreterRegistry)
+    {
+        $this->interpreterRegistry = $interpreterRegistry;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function interpret(Concrete $object, $value, Mapping $map, $data, DefinitionInterface $definition, $params, $configuration)
+    {
+        Assert::isArray($value, 'IteratorInterpreter can only be used with array values');
+
+        $interpreter = $configuration['interpreter'];
+
+        foreach ($value as &$val) {
+            $interpreterObject = $this->interpreterRegistry->get($interpreter['type']);
```

Comment:
Why fetch this each time, you can move it outside of the loop.

Reply:
done